### PR TITLE
[Improvement-5147][Service]This judgment always returns true

### LINKED ISSUE #5147: [Improvement][Master] this expression which always evaluates "true"
*For better global communication, please give priority to using English description, thx! *

*Please review https://dolphinscheduler.apache.org/en-us/community/development/issue.html when describe an issue.*

**Describe the question**
this expression which always evaluates "true"

**What are the current deficiencies and the benefits of improvement**
this expression which always evaluates "true" ，removeTaskLogFile  method of org.apache.dolphinscheduler.service.process

**Which version of DolphinScheduler:**
 -[1.3.5]

**Describe alternatives you've considered**
public void removeTaskLogFile(Integer processInstanceId) {

        LogClientService logClient = null;

        try {
            logClient = new LogClientService();
            List<TaskInstance> taskInstanceList = findValidTaskListByProcessId(processInstanceId);

            if (CollectionUtils.isEmpty(taskInstanceList)) {
                return;
            }

            for (TaskInstance taskInstance : taskInstanceList) {
                String taskLogPath = taskInstance.getLogPath();
                if (StringUtils.isEmpty(taskInstance.getHost())) {
                    continue;
                }
                int port = Constants.RPC_PORT;
                String ip = "";
                try {
                    ip = Host.of(taskInstance.getHost()).getIp();
                } catch (Exception e) {
                    // compatible old version
                    ip = taskInstance.getHost();
                }

                // remove task log from loggerserver
                logClient.removeTaskLog(ip, port, taskLogPath);
            }
        } finally {
 
                logClient.close();
            
        }
    }

### PR DESCRIPTION
**Pull request by lyyprean**
## Purpose of the pull request
This judgment always returns true

This closes #5147

## Brief change log
Change the removeTaskLogFile method in ProcessService

### CODE REVIEW

**Review by zhuangchong — COMMENTED** [review]
I suggest to change it like this

```
LogClientService logClient = null;

  
       
        List<TaskInstance> taskInstanceList = findValidTaskListByProcessId(processInstanceId);

        if (CollectionUtils.isEmpty(taskInstanceList)) {
            return;
        }

  try {
     logClient = new LogClientService();
        for (TaskInstance taskInstance : taskInstanceList) {
            String taskLogPath = taskInstance.getLogPath();
            if (StringUtils.isEmpty(taskInstance.getHost())) {
                continue;
            }
            int port = Constants.RPC_PORT;
            String ip = "";
            try {
                ip = Host.of(taskInstance.getHost()).getIp();
            } catch (Exception e) {
                // compatible old version
                ip = taskInstance.getHost();
            }

            // remove task log from loggerserver
            logClient.removeTaskLog(ip, port, taskLogPath);
        }
    } finally {

           if (logClient != null) {
            logClient.close();
        }
        
    }
}
```

**Comment by ruanwenjun** [comment]
> I suggest to change it like this
> 
> ```
> LogClientService logClient = null;
> 
>   
>        
>         List<TaskInstance> taskInstanceList = findValidTaskListByProcessId(processInstanceId);
> 
>         if (CollectionUtils.isEmpty(taskInstanceList)) {
>             return;
>         }
> 
>   try {
>      logClient = new LogClientService();
>         for (TaskInstance taskInstance : taskInstanceList) {
>             String taskLogPath = taskInstance.getLogPath();
>             if (StringUtils.isEmpty(taskInstance.getHost())) {
>                 continue;
>             }
>             int port = Constants.RPC_PORT;
>             String ip = "";
>             try {
>                 ip = Host.of(taskInstance.getHost()).getIp();
>             } catch (Exception e) {
>                 // compatible old version
>                 ip = taskInstance.getHost();
>             }
> 
>             // remove task log from loggerserver
>             logClient.removeTaskLog(ip, port, taskLogPath);
>         }
>     } finally {
> 
>            if (logClient != null) {
>             logClient.close();
>         }
>         
>     }
> }
> ```
Agree.  By the way, if the LogClientService implements AutoCloseable, we can even use try with resource
```
try(LogClientService clientService = new ClientService){
}catch(Exception){
}
```

**Comment by CalvinKirs** [comment]
hi, I agree with @ruanwenjun ,too many finally and try statements are always confusingI suggest you change it like this.

**Comment by lyyprean** [comment]
ok

**Comment by lyyprean** [comment]
> hi, I agree with @ruanwenjun ,too many finally and try statements are always confusingI suggest you change it like this.

I have modified it, thank you

**Comment by CalvinKirs** [comment]
code style check fail, you can refer to https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html

**Comment by lyyprean** [comment]
> code style check fail, you can refer to https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html
 I see. Thank you
To compatible the old version, Try is used twice ,SonarCloud Code Analysis failed

**Review by zhuangchong — CHANGES_REQUESTED** [review]
Please modify all methods that refer to LogClientService in the code.

**Comment by lyyprean** [comment]
> > code style check fail, you can refer to https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html
> >
> >I have modified it.thank you

**Comment by lyyprean** [comment]
> > code style check fail, you can refer to https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html
> >
> >I have modified it.thank you



> Please modify all methods that refer to LogClientService in the code.

ok

**Comment by lyyprean** [comment]
> Please modify all methods that refer to LogClientService in the code.

Very good proposal, I have modified it

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 1]
```diff
@@ -711,6 +700,14 @@ private ProcessInstance constructProcessInstance(Command command, String host) {
...
                     setGlobalParamIfCommanded(processDefinition, cmdParam);
                 }
+
+                // Recalculate global parameters after rerun.
+
+                processInstance.setGlobalParams(ParameterUtils.curingGlobalParams(
+                        processDefinition.getGlobalParamMap(),
+                        processDefinition.getGlobalParamList(),
+                        commandTypeIfComplement,
+                        processInstance.getScheduleTime()));
```
Comment (zhuangchong) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java]: What problem does this part of the code solve?

Comment (lyyprean) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java]: It is useless, has been deleted, thanks for correcting

**Review by zhuangchong — APPROVED** [review]
LGTM.

**Comment by zhuangchong** [comment]
considering that it's your first contribution, I think we can get deep communiction, you can contact me through mail or add wechat(KerwinZhuang), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

**Comment by lyyprean** [comment]
> considering that it's your first contribution, I think we can get deep communiction, you can contact me through mail or add wechat(KerwinZhuang), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

Thank you, I have added your WeChat

**Pull request merged**